### PR TITLE
[SPARK-33591][SQL][FOLLOW-UP] Revise the version and doc of `spark.sql.legacy.parseNullPartitionSpecAsStringLiteral`

### PR DESCRIPTION
**Pull request by gengliangwang**
### What changes were proposed in this pull request?

Correct the version of SQL configuration `spark.sql.legacy.parseNullPartitionSpecAsStringLiteral` from 3.2.0 to 3.0.2.
Also, revise the documentation and test case.

### Why are the changes needed?

The release version in https://github.com/apache/spark/pull/31421 was wrong.

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

Unit tests

### CODE REVIEW

**Comment by gengliangwang** [comment]
cc @HyukjinKwon 
I will create two backport PRs to branch 3.1 and 3.0 once this is merged.

**Comment by HyukjinKwon** [comment]
cc @cloud-fan too

**Comment by cloud-fan** [comment]
@gengliangwang I've left more comments in the original PR, can you address them as well? thanks!

**Comment by gengliangwang** [comment]
@cloud-fan yes, done.

File: sql/core/src/test/scala/org/apache/spark/sql/execution/command/ShowPartitionsSuiteBase.scala [hunk 1]
```diff
@@ -161,7 +161,7 @@ trait ShowPartitionsSuiteBase extends QueryTest with DDLCommandTestUtils {
         sql(s"CREATE TABLE $t (col1 INT, p1 STRING) $defaultUsing PARTITIONED BY (p1)")
         sql(s"INSERT INTO TABLE $t PARTITION (p1 = null) SELECT 0")
         runShowPartitionsSql(s"SHOW PARTITIONS $t", Row("p1=null") :: Nil)
-        runShowPartitionsSql(s"SHOW PARTITIONS $t PARTITION (p1 = null)", Row("p1=null") :: Nil)
+        checkAnswer(spark.sql(s"SELECT * FROM $t"), Row(0, "null"))
```
Comment (cloud-fan) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/command/ShowPartitionsSuiteBase.scala:164]: I don't know why we put the test in `ShowPartitionsSuite`, as the correctness bug is more obvious in the table scan case. Additionally, this test suite is master only and may bring troubles during backporting.

Shall we simply put the test in `SQLQuerySuite`?

Comment (gengliangwang) [reply sql/core/src/test/scala/org/apache/spark/sql/execution/command/ShowPartitionsSuiteBase.scala:164]: Sure

**Comment by cloud-fan** [comment]
LGTM, can we update the PR title?

**Comment by gengliangwang** [comment]
> LGTM, can we update the PR title?

Done

**Comment by cloud-fan** [comment]
thanks, merging to master!

**Pull request closed without merging**